application/common/components/personnel/IdBroker.php

```diff
@@ -193,4 +195,26 @@ private function getIdBrokerClient()
         );
     }
 
+    /**
+     * Updates properties on a personnel record. At a minimum, `$properties` must
+     * contain an `'employee_id'` key.
+     *
+     * @param array $properties
+     * @throws NotFoundException
+     * @throws ServiceException
+     */
+    public function updateUser($properties)
+    {
+        $idBrokerClient = $this->getIdBrokerClient();
+
+        try {
+            $idBrokerClient->updateUser($properties);
+        } catch (ServiceException $e) {
+            if ($e->httpStatusCode == 204) {
+                throw new NotFoundException();
```

Comment:
Wouldn't an attempt to update a non-existent user return a `404` HTTP status code (not a `204`)?

Reply:
Hmm, you're right. I remember now that we had long conversations about 404 vs 204 in the API design, but it was long enough ago that I don't remember the full details.